src/fdc/D88DiskImage.cc

```diff
@@ -0,0 +1,350 @@
+{
+	assert(side < 2);
+	if (singleSided && side) {
+		// NoExists Side
+		output.clear(RawTrack::STANDARD_SIZE);
+        return;
+	}
+
+	if (!readD88Track(track, side))
+	{
+		// Track Not Found
+		output.clear(RawTrack::STANDARD_SIZE);
+        return;
+	}
+
+	// # preamble (146)
+	//
+	// GAP4a   0x4E :  80
+	// SYNC    0x00 :  12
+	// IAM     0xC2 :   3
+	//         0xFC :   1
+	// GAP1    0x4E :  50
+	//
+	// # sector (62 + datasize + gap3)
+	//
+	// SYNC    0x00 :  12
+	// IDAM    0xA1 :   3
+	//         0xFE :   1
+	// CHRN      xx :   4
+	// CRC       xx :   2
+	// GAP2    0x4E :  22
+	//
+	// SYNC    0x00 :  12
+	// DAM     0xA1 :   3
+	//         0xFB :   1 (DDAM:0xF8)
+	// data      xx : 128 << N (512)
+	// CRC       xx :   2
+	// GAP3    0x4E :  84 <- 512 byte 9 sec
+	//
+	// # postamble
+	// GAP4b   0x4E : 182 <- 512 byte 9 sec
+	int trackLen = 146;
+	for(const auto& sct: trackSectors)
+	{
+		trackLen += 62 + 84 + sct.sectorSize;
+	}
+	trackLen += 182;
+	output.clear(trackLen);
+
+	int idx = 0;
+	auto write = [&](unsigned n, uint8_t value) {
+		repeat(n, [&] { output.write(idx++, value); });
+	};
+	auto trackDataSpan = std::span{trackData};
+
+	// preamble
+	write(80, 0x4E); // gap4a
+	write(12, 0x00); // sync
+	write( 3, 0xC2); // index mark (1)
+	write( 1, 0xFC); //            (2)
+	write(50, 0x4E); // gap1
+
+	// sectors
+	for(const auto& sct: trackSectors)
+	{
+		if (sct.status == D88_STATUS_NO_ADDRESS_MARK)
+		{
+			continue;
+		}
+
+		write(12, 0x00); // sync
+
+		write( 3, 0xA1);                 	// addr mark (1)
+		output.write(idx++, 0xFE, true); 	//           (2) add idam
+		output.write(idx++, sct.c); 		// C: Cylinder number
+		output.write(idx++, sct.h);  		// H: Head Address
+		output.write(idx++, sct.r); 		// R: Record
+		output.write(idx++, sct.n);  		// N: Number (length of sector: 512 = 128 << 2)
+		uint16_t addrCrc = output.calcCrc(idx - 8, 8);
+		if (sct.status == D88_STATUS_ID_CRC_ERROR)
+		{
+			addrCrc = ~addrCrc;
+		}
+		output.write(idx++, narrow_cast<uint8_t>(addrCrc >> 8));   // CRC (high byte)
+		output.write(idx++, narrow_cast<uint8_t>(addrCrc & 0xff)); //     (low  byte)
+		
+		write(22, 0x4E); // gap2
+
+		if (sct.status == D88_STATUS_NO_DATA_MARK)
+		{
+			continue;
+		}
+
+		write(12, 0x00); // sync
+		write( 3, 0xA1); // data mark (1)
+		if (sct.deleteFlag == 0)
+		{
+			write( 1, 0xFB); //           (2)
+		}
+		else
+		{
+			write( 1, 0xF8); //           (2)
+		}
+		output.writeBlock(idx, trackDataSpan.subspan(sct.dataOffset, sct.dataSize));
+		idx += sct.sectorSize;
+
+		uint16_t dataCrc = output.calcCrc(idx - (sct.sectorSize + 4), sct.sectorSize + 4);
+		if (sct.status == D88_STATUS_DATA_CRC_ERROR)
+		{
+			dataCrc = ~dataCrc;
+		}
+		output.write(idx++, narrow_cast<uint8_t>(dataCrc >> 8));   // CRC (high byte)
+		output.write(idx++, narrow_cast<uint8_t>(dataCrc & 0xff)); //     (low  byte)
+
+		write(84, 0x4E); // gap3
+	}
+	//write(182, 0x4E); // gap4b

```

Comment:
I think it's important to also write these last 182 bytes. We should not leave any uninitialized data in the track.
* With normal sector read commands you cannot read these bytes.
* But you can with the WD2793 "Read Track" command.